rip: add page (#8121)
tldr-pages/tldr: pages/common/rip.md

# rip

> Remove files or directories by sending them to the graveyard, allowing for them to be recovered.
> More information: <https://github.com/nivekuil/rip>.

> - Remove files or directories from arbitrary locations and place them in the graveyard:

Reviewer: ```suggestion
- Remove files or directories from specified locations and place them in the graveyard:
```

`rip {{path/to/file_or_directory}} {{path/to/another/file_or_directory}}`

- Interactively remove files or directories, with a prompt before every removal:

> `rip -i {{path/to/file_or_directory}} {{path/to/another/file_or_directory}}`

Reviewer 2: We prefer long arguments if available
```suggestion
`rip --inspect {{path/to/file_or_directory}} {{path/to/another/file_or_directory}}`
```

Reviewer 2: @superatomic this new description is not clear about the `prompt`.

Author: Ah yes, you're right about that. How about this:
```markdown
- Interactively [i]nspect files or directories and prompt for confirmation before removing them:

`rip -i {{path/to/file_or_directory}} {{path/to/another/file_or_directory}}`
```

Author: Okay, that's fair; I'll make the changes. Should I also change the description to my new suggestion?

Reviewer 3: I like them both equally. Take whichever you prefer.

Author: Got it! I think I'm going to keep it as is.

> - List all files in the graveyard that were somewhere in the current directory:

Reviewer: ```suggestion
- List all files in the graveyard that were originally somewhere in the current directory:
```

> - List all files and directories in the graveyard that were originally within the current directory:

Reviewer: ```suggestion
- List all files and directories in the graveyard that were originally within the current working directory:
```

Author: Personally, I feel like this line is already really long and it might not be worth adding "working" to be more clear. If you disagree (or if you feel that line length doesn't matter at all) I can commit this suggestion.

Reviewer: No, it is all fine, I tend to leave further edits which I am not bothered about if they don't get committed after approving!

> `rip -s`

Reviewer 2: ```suggestion
`rip --seance`
```

- Permanently delete every file and directory in the graveyard:

> `rip -d`

Reviewer 2: ```suggestion
`rip --decompose`
```

> - Undo the last removal:

Reviewer: ```suggestion
- Replace the files which were affected by the most recent removal:
```

`rip -u`

> - Undo all removals that are listed by `rip -s`:

Reviewer: ```suggestion
- Replace all files that are listed by `rip -s`:
```

Author: The `-u` stands for [U]ndo, so it might be worth it to keep it as "Undo". If you still want to get rid of undo, "Put back" might be more specific than "Replace".

Reviewer: In my opinion, using replace is a clearer way of describing the command. However I like the usage of `Put back` feel free to change to that.

Author: That sounds fair. These changes have been made.

> `rip -u`

Reviewer 2: ```suggestion
`rip --unbury`
```

> - Put back every file and directory that is listed by `rip -s`:

Reviewer 2: ```suggestion
- Put back every file and directory that is listed by `rip --seance`:
```

> `rip -su`

Reviewer 2: ```suggestion
`rip --seance --unbury`
```

Author: In my opinion, `--seance --unbury` is way to verbose compared to `-su`, and additionally, the name `--seance` is not the most helpful name for understanding what the option does. I think it would be clearer to leave it in the short form.

Of course, if you still disagree, I can make the change.

Reviewer 3: I do agree, that the convention to use long options is always great, but it makes maintaining a lot easier. Also, most of the time it's easy to quickly find out the short version.
